### PR TITLE
Add legacy support (Python 3.7 & 3.8) in Sterzo

### PR DESCRIPTION
**Pull request by tensorturtle**
[`importlib.resources.files` in `sterzo.py`](https://github.com/zacharyedwardbull/pycycling/blob/ec828e863ffdfe9d3fdfc65b33981f9889f830bb/pycycling/sterzo.py#L25C36-L25C36) was newly introduced in Python 3.9.
I needed to use this library on Python 3.8, so I implemented a version check to use the correct importlib API. [More info at python docs](https://docs.python.org/3/library/importlib.resources.html)

The only side effect is that the `sterzo-challenge-codes.dat` file had to be moved out of the `data` directory.

### CODE REVIEW

**Comment by tensorturtle** [comment]
I tested the change on Python 3.8 and Python 3.10 with success.

File: pycycling/sterzo.py [hunk 1]
```diff
@@ -22,10 +23,19 @@ async def enable_steering_measurement_notifications(self):
         await self._activate_steering_measurements()
 
     async def _activate_steering_measurements(self):
-        with importlib.resources.files(__package__).joinpath('data/sterzo-challenge-codes.dat').open('rb') as fp:
-            fp.seek(self._latest_challenge * 2, 1)
-            code_1 = int.from_bytes(fp.read(1), 'little')
-            code_2 = int.from_bytes(fp.read(1), 'little')
+        # moved .dat file out of 'data' because accessing directories not possible for importlib.resources.path
```
Comment (zacharyedwardbull) [inline pycycling/sterzo.py:27]: I think we can keep the data directory, we just need to add `__init__.py` to make it a Python package

File: pycycling/sterzo.py [hunk 2]
```diff
@@ -22,10 +23,19 @@ async def enable_steering_measurement_notifications(self):
...
+        # importlib.resources.path is deprecated since 3.11
+        if sys.version_info >= (3,11):
+            with importlib.resources.files(__package__).joinpath('sterzo-challenge-codes.dat').open('rb') as fp:
+                fp.seek(self._latest_challenge * 2, 1)
+                code_1 = int.from_bytes(fp.read(1), 'little')
+                code_2 = int.from_bytes(fp.read(1), 'little')
+        else: # legacy support < 3.9
+            with importlib.resources.path(__package__, 'sterzo-challenge-codes.dat') as file_path:
```
Comment (zacharyedwardbull) [inline pycycling/sterzo.py]: Is there a reason not to use `importlib.resources.open_binary`?

File: pycycling/sterzo.py [hunk 3]
```diff
@@ -22,10 +23,19 @@ async def enable_steering_measurement_notifications(self):
...
+                with open(file_path, 'rb') as fp:
+                    fp.seek(self._latest_challenge * 2, 1)
```
Comment (zacharyedwardbull) [inline pycycling/sterzo.py]: Can we refactor this to remove the duplication of these lines?

**Review by zacharyedwardbull — COMMENTED** [review]
Hey! Thanks for spotting this lack of compatibility with Python 3.7/3.8 and suggesting this fix. It looks good, I've just left a few comments.

File: pycycling/sterzo.py (hunk 1 above)
Comment (tensorturtle) [reply pycycling/sterzo.py:27]: Do you mean putting a `__init__.py` to the `data` directory like so?
```
├── pycycling
       ├── sterzo.py
       ├── data
       │   ├── __init__.py
       │   └── sterzo-challenge-codes.dat
```

File: pycycling/sterzo.py (hunk 2 above)
Comment (tensorturtle) [reply pycycling/sterzo.py]: Will be using that instead.

File: pycycling/sterzo.py (hunk 1 above)
Comment (zacharyedwardbull) [reply pycycling/sterzo.py:27]: Yep! Can be completely empty I believe.

**Comment by tensorturtle** [comment]
Putting `sterzo-challenges-codes.dat` into a `data` directory with `__init__.py` does not work:

Using: 
```
with importlib.resources.open_binary(__package__, 'data/sterzo-challenge-codes.dat') as fp:
```
Throws:
```
  File "/home/tensorturtle/Repos/pycycling/pycycling/sterzo.py", line 34, in _activate_steering_measurements
    with importlib.resources.open_binary(__package__, 'data/sterzo-challenge-codes.dat') as fp:
  File "/usr/lib/python3.10/importlib/resources.py", line 42, in open_binary
    resource = _common.normalize_path(resource)
  File "/usr/lib/python3.10/importlib/_common.py", line 34, in normalize_path
    raise ValueError(f'{path!r} must be only a file name')
ValueError: 'data/sterzo-challenge-codes.dat' must be only a file name
```

I think this limitation is what the docs are referring to when it says:

> The main drawback of these functions is that they do not support directories: they assume all resources are located directly within a package.
https://docs.python.org/3/library/importlib.resources.html#deprecated-functions

**Comment by zacharyedwardbull** [comment]
Since we've created a subpackage, you need to add a `import pycycling.data` to the imports then replace `__package__` with `pycycling.data` in the call.

**Comment by tensorturtle** [comment]
Thank you for guiding me. Please see latest commit

**Comment by tensorturtle** [comment]
I tested both branches

**Comment by tensorturtle** [comment]
Fix #28

**Pull request merged**